lib/active_model_serializers/adapter/json_api/resource_identifier.rb

```diff
@@ -20,6 +20,7 @@ def as_json
         private
 
         def type_for(serializer)
+          return serializer._type.call(serializer.object) if serializer._type.is_a?(Proc)
```

Comment:
I don't think so, but since we fundamentally do not care whether it is a `Proc` or not as long as it is a callable, we might as well adopt duck typing here.